pinot-minion/src/main/java/org/apache/pinot/minion/BaseMinionStarter.java

```diff
@@ -102,7 +102,12 @@ public void init(PinotConfiguration config)
     setupHelixSystemProperties();
     _helixManager = new ZKHelixManager(helixClusterName, _instanceId, InstanceType.PARTICIPANT, zkAddress);
     MinionTaskZkMetadataManager minionTaskZkMetadataManager = new MinionTaskZkMetadataManager(_helixManager);
-    _taskExecutorFactoryRegistry = new TaskExecutorFactoryRegistry(minionTaskZkMetadataManager);
+    int endReplaceSegmentsSocketTimeoutMs =
+        _config.getProperty(CommonConstants.Minion.END_REPLACE_SEGMENTS_SOCKET_TIMEOUT_MS_KEY) != null
```

Comment:
Is there any existing method like getOrDefault that you can use to make it clean?

Reply:
Good point, using the getOrDefault function instead.